src/libcrun/utils.c

```diff
@@ -1225,14 +1225,35 @@ check_access (const char *path)
 }
 
 const char *
-find_executable (const char *executable_path)
+find_executable (const char *executable_path, const char *cwd)
 {
+  cleanup_free char *cwd_executable_path = NULL;
   cleanup_free char *tmp = NULL;
   char path[PATH_MAX + 1];
   int last_error = ENOENT;
   char *it, *end;
   int ret;
 
+  if (executable_path[0] == '.')
```

Comment:
Should you check to make sure `len(executable_path) >0`?

Reply:
in that case executable_path[0] would be `\0`

Reply:
What about if executable_path == null

Reply:
that should not happen as the only caller checks for it, but in any case to be safer I've added a new check now